tests/functional/refinement/divb_refinement.py

```
@@ -0,0 +1,246 @@
+# 0); the evolved+reconstructed coarse floor is ~2e-7 for both modes, and the fine level sits
+# at ~2x that (a refinement-boundary truncation factor), NOT at the O(B/dx)~0.1 a misclassified
+# shared face would give. The contract: fine <= REL_TOL * max(coarse-floor, ABS_TOL) AND fine
+# within REL_TOL of the order-0 baseline (isolates the operator from the init floor). ABS_TOL
+# keeps a non-amplifying operator from tripping on the small coarse-floor noise.
+ABS_TOL = 1e-4  # floor below which divB is "already zero" (truncation, not a bug)
+REL_TOL = 5.0   # fine must not exceed REL_TOL x (coarse floor) nor REL_TOL x (order-0 fine)
+
+
+# Shared Harris double-sheet init (both modes). Bx is a function of y ONLY (double tanh,
+# sheets at 0.3Ly and 0.7Ly), By = Bz_tangential = 0 => divB = 0 to machine precision.
+def _S(y, y0, l):
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)